backend/src/apiserver/resource/pod_fake.go

```diff
@@ -0,0 +1,79 @@
+package resource
```

Comment:
Why do we need this? Client side mocks are really difficult to maintain.

Reply:
this is following the same as other fakes in the same folder. It's used for testing the Retry logic.